transport/tchannel/error.go

```diff
@@ -22,12 +22,56 @@ package tchannel
 
 import (
 	"context"
+	"errors"
 
 	"github.com/uber/tchannel-go"
 	"go.uber.org/yarpc/api/transport"
 	"go.uber.org/yarpc/yarpcerrors"
 )
 
+// GetResponseErrorMeta extracts the TChannel specifc error response information
+// from an error. Returns nil if no information is available.
+//
+// This API is experimental and subject to change or break at anytime.
+func GetResponseErrorMeta(err error) *ResponseErrorMeta {
+	if err == nil {
+		return nil
+	}
+
+	var meta *ytchanError
+	if !errors.As(err, &meta) {
+		return nil
+	}
+	return &ResponseErrorMeta{
+		Code: meta.code,
+	}
+}
+
+// ResponseErrorMeta exposes TChannel specific information from an error.
+//
+// This API is experimental and subject to change or break at anytime.
+type ResponseErrorMeta struct {
+	Code tchannel.SystemErrCode
+}
+
+// private error for propagating transparently
+type ytchanError struct {
+	code tchannel.SystemErrCode
+}
+
+func (y *ytchanError) Error() string { return "" } // no-op, this should never be used directly
```

Comment:
Yup, good call, this would cause strange behaviour to direct `Unwrap` calls. 
Storing the error will also make constructing the error more straightforward/readable too 👍